Tests/SentryTests/Integrations/SentryPerformanceTrackerTests.swift

```diff
@@ -0,0 +1,219 @@
+        
+        func getSut(initSDK: Bool = false) -> SentryPerformanceTracker {
+            if initSDK {
+                SentrySDK.start { options in
+                    options.dsn = SentryPerformanceTrackerTests.dsnAsString
+                    options.debug = true
+                    options.diagnosticLevel = SentryLevel.debug
+                }
+            }
+            return  SentryPerformanceTracker()
+        }
+
+        func getSpans(tracker: SentryPerformanceTracker) -> [SpanId: Span] {
+            let result = Dynamic(tracker).spans as [SpanId: Span]?
+            return result!
+        }
+        
+        func getStack(tracker: SentryPerformanceTracker) -> [Span] {
+            let result = Dynamic(tracker).activeStack as [Span]?
+            return result!
+        }
+    }
+    
+    private var fixture: Fixture!
+    
+    override func setUp() {
+        
+        fixture = Fixture()
+    }
+    
+    override func tearDown() {
+        SentrySDK.close()
+    }
+    
+    func testSingleton() {
+        XCTAssertEqual(SentryPerformanceTracker.shared(), SentryPerformanceTracker.shared())
+    }
+    
+    func testStartSpan_CheckScopeSpan() {
+        let tracker = fixture.getSut(initSDK: true)
+        let spanid = tracker.startSpan(withName: fixture.someTransaction, operation: fixture.someOperation)
+        let spans: [SpanId: Span] = fixture.getSpans(tracker: tracker)
+        
+        let transaction = spans[spanid] as! SentryTracer
+        
+        let scopeSpan = SentrySDK.currentHub().scope.span
+        
+        XCTAssert(scopeSpan === transaction)
+        XCTAssertTrue(transaction.waitForChildren)
+    }
+    
+    func testStartSpan_ScopeAlredyWithSpan() {
+        let tracker = fixture.getSut(initSDK: true)
+
+        let firstTransaction = SentrySDK.startTransaction(name: fixture.someTransaction, operation: fixture.someOperation, bindToScope: true)
+        let spanid = tracker.startSpan(withName: fixture.someTransaction, operation: fixture.someOperation)
+        let spans: [SpanId: Span] = fixture.getSpans(tracker: tracker)
+        
+        let transaction = spans[spanid]
+        let scopeSpan = SentrySDK.currentHub().scope.span
+        
+        XCTAssert(scopeSpan !== transaction)
+        XCTAssert(scopeSpan === firstTransaction)
+    }
+  
+    func testStartSpan_WithActiveSpan() {
+        let tracker = fixture.getSut()
+        let spanid = tracker.startSpan(withName: fixture.someTransaction, operation: fixture.someOperation)
+        
+        tracker.pushActiveSpan(spanid)
+        
+        let childSpanId = tracker.startSpan(withName: fixture.someTransaction, operation: fixture.someOperation)
+        let spans: [SpanId: Span] = fixture.getSpans(tracker: tracker)
+        
+        let transaction = spans[spanid]
+        let childSpan = spans[childSpanId]
+ 
+        let children = Dynamic(transaction).children as [Span]?
+        
+        XCTAssert(children!.first === childSpan)
```

Comment:
`l`: This would improve the test slighly.
```suggestion
        XCTAssertEqual(1, children?.count)
        XCTAssert(children!.first === childSpan)
```